test/xpu_api/numerics/numeric.ops/inner.product/xpu_inner_product_comp.pass.cpp

```diff
@@ -14,144 +14,109 @@
-  test<input_iterator<const int *>, forward_iterator<const int *>,
-       KernelTest2>();
-  test<input_iterator<const int *>, bidirectional_iterator<const int *>,
-       KernelTest3>();
-  test<input_iterator<const int *>, random_access_iterator<const int *>,
-       KernelTest4>();
-  test<input_iterator<const int *>, const int *, KernelTest5>();
+#if !PSTL_USE_DEBUG
+    test<forward_iterator<const int*>, input_iterator<const int*>>();
+    test<forward_iterator<const int*>, forward_iterator<const int*>>();
+    test<forward_iterator<const int*>, bidirectional_iterator<const int*>>();
+    test<forward_iterator<const int*>, random_access_iterator<const int*>>();
+    test<forward_iterator<const int*>, const int*>();
 
-  test<forward_iterator<const int *>, input_iterator<const int *>,
-       KernelTest6>();
-  test<forward_iterator<const int *>, forward_iterator<const int *>,
-       KernelTest7>();
-  test<forward_iterator<const int *>, bidirectional_iterator<const int *>,
-       KernelTest8>();
-  test<forward_iterator<const int *>, random_access_iterator<const int *>,
-       KernelTest9>();
-  test<forward_iterator<const int *>, const int *, KernelTest10>();
+    test<bidirectional_iterator<const int*>, input_iterator<const int*>>();
+    test<bidirectional_iterator<const int*>, forward_iterator<const int*>>();
+    test<bidirectional_iterator<const int*>, bidirectional_iterator<const int*>>();
+    test<bidirectional_iterator<const int*>, random_access_iterator<const int*>>();
+    test<bidirectional_iterator<const int*>, const int*>();
 
-  test<bidirectional_iterator<const int *>, input_iterator<const int *>,
-       KernelTest11>();
-  test<bidirectional_iterator<const int *>, forward_iterator<const int *>,
-       KernelTest12>();
-  test<bidirectional_iterator<const int *>, bidirectional_iterator<const int *>,
-       KernelTest13>();
-  test<bidirectional_iterator<const int *>, random_access_iterator<const int *>,
-       KernelTest14>();
-  test<bidirectional_iterator<const int *>, const int *, KernelTest15>();
+    test<random_access_iterator<const int*>, input_iterator<const int*>>();
+    test<random_access_iterator<const int*>, forward_iterator<const int*>>();
+    test<random_access_iterator<const int*>, bidirectional_iterator<const int*>>();
+    test<random_access_iterator<const int*>, random_access_iterator<const int*>>();
+    test<random_access_iterator<const int*>, const int*>();
 
-  test<random_access_iterator<const int *>, input_iterator<const int *>,
-       KernelTest16>();
-  test<random_access_iterator<const int *>, forward_iterator<const int *>,
-       KernelTest17>();
-  test<random_access_iterator<const int *>, bidirectional_iterator<const int *>,
-       KernelTest18>();
-  test<random_access_iterator<const int *>, random_access_iterator<const int *>,
-       KernelTest19>();
-  test<random_access_iterator<const int *>, const int *, KernelTest20>();
+    test<const int*, input_iterator<const int*>>();
```

Comment:
I would also leave the cases with raw pointers because they look unique comparing to the rest ones:  here and in the other test.

Reply:
Thanks for your comments. The cases with raw pointers significantly increase running time and can we just disable them?

Reply:
I see. Let's disable them if so.